pkg/api/controller.go

```diff
@@ -333,6 +333,21 @@ func (c *Controller) GetPhysicalAddress(w http.ResponseWriter, r *http.Request,
 		PhysicalAddress: StringPtr(qk.Format()),
 		Token:           StringValue(token),
 	}
+
+	if params.Presign != nil && *params.Presign {
+		// generate a pre-signed PUT url for the given request
+		preSignedURL, err := c.BlockAdapter.GetPreSignedURL(ctx, block.ObjectPointer{
+			StorageNamespace: repo.StorageNamespace,
+			Identifier:       address,
+			IdentifierType:   block.IdentifierTypeRelative,
+		}, block.PreSignModeWrite)
+		if err != nil {
+			writeError(w, r, http.StatusInternalServerError, err)
+			return
+		}
+		response.PresignedUrl = StringPtr(preSignedURL)
```

Comment:
Nit: I think
```suggestion
		response.PresignedUrl = &preSignedURL
```
is more readable as well as more efficient.